src/main/java/net/greenfieldmc/core/advancedbuild/handlers/SwitchInteraction.java

```diff
@@ -7,59 +7,93 @@
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.TextComponent;
 import org.bukkit.Material;
+import org.bukkit.block.Block;
 import org.bukkit.block.BlockFace;
 import org.bukkit.block.data.FaceAttachable;
 import org.bukkit.block.data.type.Switch;
 import org.bukkit.event.Event;
 import org.bukkit.event.player.PlayerInteractEvent;
 
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
 public class SwitchInteraction extends InteractionHandler {
 
+    private final Map<UUID, Boolean> sessions = new HashMap<>();
+
     public SwitchInteraction(IWorldEditService worldEditService, ICoreProtectService coreProtectService) {
         super(worldEditService, coreProtectService, (InteractPredicate) (event) -> {
-                    var player = event.getPlayer();
-                    var mainHand = player.getInventory().getItemInMainHand().getType();
-                    return mainHand != Material.AIR &&
-                            event.getClickedBlock() != null &&
-                            mainHand.isBlock() && mainHand.createBlockData() instanceof Switch;
-                }
-        );
+            var player = event.getPlayer();
+            var mainHand = player.getInventory().getItemInMainHand().getType();
+            var clicked = event.getClickedBlock();
+            if (clicked == null) return false;
+            return (mainHand == Material.AIR && isSwitch(clicked.getType()))
+                    || (mainHand.isBlock() && isSwitch(mainHand));
+        });
     }
 
     @Override
     public TextComponent getInteractionDescription() {
-        return Component.text("Allow the unnatural placement of blocks that are considered switches. (buttons, levers, etc)");
+        return Component.text("Allow unnatural placement and state control of switches (buttons, levers, etc). Shift-right-click to toggle, shift-place to use saved state.");
     }
 
     @Override
     public TextComponent getInteractionUsage() {
-        return Component.text("Shift and right click to place a switch type block against the blockface you clicked.");
+        return Component.text("Shift and right click to place or toggle a switch type block. Shift-place to use saved powered state.");
```

Comment:
This should be more descriptive. Explaining the difference between an empty hand and a non-empty hand would be useful.